### PR TITLE
[FE] Make EditableCell in table responsive

### PR DESCRIPTION
**Pull request by akash-joshi**
## 🌟 What is the purpose of this PR?

The existing demo for table block doesn't work well on mobile. All the editable cells on the table have complete vertical scroll with each letter appearing on a new line. This PR fixes this by allowing each item in a cell to take the maximum width it can without causing a wrap.

## ⚠️ Known issues

- Since we're using `textarea`s for editable cells, we can't have it auto-wrap the content present inside it, ie, `width: max-content` won't work. Hence, I'm relying on using `max-width`s (which textarea does listen to) as a way to have sensible default widths for both desktop and mobile.

## 🔍 What does this change?

Check GitHub comments below

## 🔗 Related links

- [Improve table block rendering on mobile](https://app.asana.com/0/1201668052739245/1201303965906632/f) 

## ❓ How to test this?

- Checkout the branch and start the dev version of table block locally.
- Check the responsiveness of the cells in various widths.

## 📹 Demo

chrome_paZqMXGSvw
chrome_w37geNGBJz

### CODE REVIEW

File: packages/blocks/table/src/App.tsx [hunk 1]
```diff
@@ -368,7 +368,7 @@ export const App: BlockComponent<AppProps> = ({
 
   /** @todo Fix keys in iterators below to not use the index */
   return (
-    <div>
+    <div className={tw`overflow-x-auto`}>
```
Comment (akash-joshi) [inline packages/blocks/table/src/App.tsx:371]: This overflow tag allows the table to overflow well. Adding an `overflow-x` to its parent results in some columns being hidden in mobile view.

File: packages/blocks/table/src/components/EditableCell.tsx [hunk 2]
```diff
@@ -39,7 +39,7 @@ export const EditableCell: VoidFunctionComponent<EditableCellProps> = ({
     const inputEl = textAreaRef.current;
 
     inputEl.style.height = "auto";
-    inputEl.style.height = `${inputEl.scrollHeight}px`;
+    inputEl.style.height = `${inputEl.scrollHeight + 2}px`;
```
Comment (akash-joshi) [inline packages/blocks/table/src/components/EditableCell.tsx:42]: An offset of +2 px was needed to not cause over-scroll on wrap.

File: packages/blocks/table/src/components/EditableCell.tsx [hunk 3]
```diff
@@ -86,7 +86,7 @@ export const EditableCell: VoidFunctionComponent<EditableCellProps> = ({
   return (
     <textarea
       ref={textAreaRef}
-      className={tw`block resize-none w-full bg-transparent rounded border-1 border-transparent hover:border-blue-500 py-1 px-2 -mx-2 `}
+      className={tw`block resize-none w-full bg-transparent rounded border-1 border-transparent hover:border-blue-500 py-1 px-2 -mx-2 w-max max-w-[150px] md:max-w-[300px]`}
```
Comment (akash-joshi) [inline packages/blocks/table/src/components/EditableCell.tsx:89]: Since we're using textareas for editable cells, we can't have it auto-wrap the content present inside it, ie, width: max-content won't work. Hence, I'm relying on using max-widths (which textarea does listen to) as a way to have sensible default widths for both desktop and mobile.

File: packages/blocks/table/src/mockData/mockData.ts [hunk 4]
```diff
@@ -116,34 +113,71 @@ const people: Person[] = [
...
     __linkedData: {
-      entityTypeId: "Person",
       aggregate: {
-        itemsPerPage: 3,
-        multiSort: [
-          {
-            field: "name",
-          },
-        ],
+        itemsPerPage: 5,
```
Comment (akash-joshi) [inline packages/blocks/table/src/mockData/mockData.ts:119]: (Maybe) Unrelated Change: Updating the mock data for the table block to match the data present in the hub demo.

The previous mock data was broken so this change is needed to test the block well.

File: packages/blocks/table/src/devRenderer.tsx [hunk 5]
```diff
@@ -2,187 +2,24 @@
  * webpack-dev-server entry point for debugging.
  * This file is not bundled with the library during the build process.
  */
-import React, { useCallback, useMemo, useState } from "react";
+import React from "react";
 import ReactDOM from "react-dom";
 import { tw } from "twind";
 
-import {
```
Comment (CiaranMn) [inline packages/blocks/table/src/devRenderer.tsx:9]: Once refactored for the correct links approach we can replace all this with `MockBlockDock`

**Review by CiaranMn — APPROVED** [review]
Thanks Akash, looks good

File: packages/blocks/table/src/devRenderer.tsx (hunk 5 above)
Comment (akash-joshi) [reply packages/blocks/table/src/devRenderer.tsx:9]: Good flag. Makes sense
